### ISSUE TITLE
`Envelope.from_signable` gives a wrong impression about `Signable`

### ISSUE THREAD

**Issue by lukpueh** [in-toto/in-toto#621]
**Description of issue or feature request**:

[`Envelope.from_signable`](https://github.com/in-toto/in-toto/blob/6698e7a5fcb37862c6876fa2047d498ad1145941/in_toto/models/metadata.py#L145-L147) uses `Signable` as base class for signed things (Link, Layout), and ignores its [original purpose](https://github.com/in-toto/in-toto/blob/6698e7a5fcb37862c6876fa2047d498ad1145941/in_toto/models/common.py#L47-L50):
>  """Objects with base class Signable are to be included in a Metablock class
    to be signed (hence the name). They provide a `signable_bytes` property
    used to create deterministic signatures."""

This `signable_bytes` property is actually a shortcut to the canonical JSON representation of a Signable object, which should not be used with DSSE (the whole point about DSSE is to avoid canonicalisation; also see [this comment](https://github.com/theupdateframework/python-tuf/issues/2093#issuecomment-1240463915)). This works because `Envelope.from_signable` just ignores `signable_bytes` and uses an alternative (non-canonical) bytes representation of Signable, which is better suited for its purpose.

Still `Envelope.from_signable` gives the impression that Metablock or Envelope payloads must subclass `Signable` (see https://github.com/in-toto/in-toto/pull/570#discussion_r1305696229).

**Current behavior**:

- `Envelope.from_signable` gives the impression that payloads need to subclass `Signable` 

**Expected behavior**:

- `Envelope.from_signable` should make it clear why it (mis)uses `Signable`

kudos to @chasen-bettinger for asking the right questions

**Comment by adityasaky** [comment]
I agree with clarifying this, I think in addition to the `from_signable` aspect, the confusion also arises because we're going from the predicate (legacy links) being signed to a wrapper around the predicate (Statement) being signed.
